task/golang-test/0.3/golang-test.yaml

```diff
@@ -0,0 +1,131 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: golang-test
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: Testing
+    tekton.dev/tags: test
+    tekton.dev/displayName: "golang test"
+    tekton.dev/platforms: "linux/arm64,linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This is a Golang task to test Go projects.
+
+  params:
+  - name: package
+    description: package (and its children) under test
+  - name: packages
+    description: "packages to test (default: ./...)"
+    default: "./..."
+  - name: context
+    description: path to the directory to use as context.
+    default: "."
+  - name: image
+    description: docker image to run the task
+    default: "docker.io/library/golang"
+  - name: version
+    description: golang version to use for tests
+    default: "latest"
+  - name: flags
+    description: flags to use for the test command
+    default: -race -cover -v
+  - name: GOOS
+    description: "running program's operating system target"
+    default: linux
+  - name: GOARCH
+    description: "running program's architecture target"
+    default: amd64
+  - name: GO111MODULE
+    description: "value of module support"
+    default: auto
+  - name: GOCACHE
+    description: "Go caching directory path"
+    default: ""
+  - name: GOMODCACHE
+    description: "Go mod caching directory path"
+    default: ""
+  - name: CGOENABLED
+    description: enable cgo 
+  - name: GOPRIVATE
+    description: "list of go private modules"
+  - name: GOPROXY
```

Comment:
According to https://go.dev/ref/mod#goproxy-protocol the default value should be "off" or "direct"

Reply:
off/ direct are valid values. proxy is enabled by default. Ive added default values from my unedited environment